### PR TITLE
Basic Json Rpc batch requests support

### PR DESCRIPTION
**Pull request by AlanVerbner**
## Description

While testing using Mist I've found that there are a lot batch requests being performed even if this is not stated it must be supported by clients in documentation. 

After implementing it, Mist stopped throwing errors and noticed some fixes. For example: accounts and balances are updated when added or mining.

### Notes
- It's a basic implementation but seems to be working as parity does. 
- It doesn't check if two requests with same Id are sent in the same batch request (parity doesn't neither). Same case if the items within the batch array are the same
- Does not implement any limit / restriction on the amount of items a batch request might contain. Maybe we need to create a new task for this

### CODE REVIEW

File: src/main/scala/io/iohk/ethereum/jsonrpc/http/JsonRpcHttpServer.scala [hunk 1]
```diff
@@ -52,6 +55,10 @@ class JsonRpcHttpServer(jsonRpcController: JsonRpcController, config: JsonRpcHtt
     complete(jsonRpcController.handleRequest(request))
   }
 
+  private def handleBatchRequest(requests: Seq[JsonRpcRequest]) = {
+    complete(Future.sequence(requests.map(request => jsonRpcController.handleRequest(request))))
```
Comment (rtkaczyk) [inline src/main/scala/io/iohk/ethereum/jsonrpc/http/JsonRpcHttpServer.scala]: I wonder if should synchronise those calls in case the caller expects them to be performed in the given order? IMO, finding this out is outside the scope of this PR.

File: src/main/scala/io/iohk/ethereum/jsonrpc/http/JsonRpcHttpServer.scala [hunk 2]
```diff
@@ -34,6 +35,8 @@ class JsonRpcHttpServer(jsonRpcController: JsonRpcController, config: JsonRpcHtt
   val route: Route = {
     (pathEndOrSingleSlash & post & entity(as[JsonRpcRequest])) { request =>
       handleRequest(request)
+    } ~ (pathEndOrSingleSlash & post & entity(as[Seq[JsonRpcRequest]])) { request =>
+      handleBatchRequest(request)
```
Comment (rtkaczyk) [inline src/main/scala/io/iohk/ethereum/jsonrpc/http/JsonRpcHttpServer.scala]: We could avoid repetition here:

```
    (pathEndOrSingleSlash & post) {
      entity(as[JsonRpcRequest]) { request =>
        handleRequest(request)
      } ~
      entity(as[Seq[JsonRpcRequest]]) { request =>
        handleBatchRequest(request)
      }
    }
```

File: src/main/scala/io/iohk/ethereum/jsonrpc/http/JsonRpcHttpServer.scala (hunk 1 above)
Comment (LukasGasior1) [inline src/main/scala/io/iohk/ethereum/jsonrpc/http/JsonRpcHttpServer.scala]: I think the order is maintained here, isn't it?
